finance_dl/amazon.py

```diff
@@ -63,30 +63,34 @@ def CONFIG_amazon():
 
 logger = logging.getLogger('amazon_scrape')
 
-netloc_re = r'^([^\.@]+\.)*amazon.com$'
 
-
-def check_url(url):
-    result = urllib.parse.urlparse(url)
-    if result.scheme != 'https' or not re.fullmatch(netloc_re, result.netloc):
-        raise RuntimeError('Reached invalid URL: %r' % url)
+class Domain:
+    COM = 'com'
+    CO_UK = 'co.uk'
 
 
 class Scraper(scrape_lib.Scraper):
-    def __init__(self, credentials, output_directory, **kwargs):
+    def __init__(self, credentials, output_directory, amazon_domain=Domain.COM, **kwargs):
         super().__init__(**kwargs)
         self.credentials = credentials
         self.output_directory = output_directory
         self.logged_in = False
+        self.amazon_domain = amazon_domain
+
+    def check_url(self, url):
+        netloc_re = r'^([^\.@]+\.)*amazon.' + self.amazon_domain + '$'
+        result = urllib.parse.urlparse(url)
+        if result.scheme != 'https' or not re.fullmatch(netloc_re, result.netloc):
+            raise RuntimeError('Reached invalid URL: %r' % url)
 
     def check_after_wait(self):
-        check_url(self.driver.current_url)
+        self.check_url(self.driver.current_url)
 
     def login(self):
+        logger.info('Initiating log in')
+        self.driver.get('https://www.amazon.' + self.amazon_domain)
```

Comment:
Reordered so it's easier to debug while staying logged in (was worried of Amazon trigged by too many relogins), by setting `loggen_in = True` programmatically. Previously it wasn't opening the amazon page if you did that.